src/get-started/java-binding.md

```diff
@@ -491,93 +498,297 @@ as specified in [How to Run a Service section](#how-to-run-a-service).
+transactions that the service submitted itself (e.g., in `afterCommit` method),
+as those are put into the pool.
+
+!!! note
+    Note that blocks that are created with
+    [`TestKit.createBlockWithTransactions(Iterable<TransactionMessage> transactionMessages)`][testkit-create-block]
+    will ignore in-pool transactions. As of 0.7.0, there is no way to create a block
+    that would contain both given and in-pool transactions with a single
+    method - to do that, put given transactions into TestKit transaction pool
+    with [`Node.submitTransaction(RawTransaction rawTransaction)`][node-submit-transaction].
+
+#### Checking the blockchain state
+
+In order to test service read operations and verify changes in blockchain state
+TestKit provides a snapshot of the current database state (i.e., the one that
+corresponds to the latest committed block). There are several ways to access
+it:
+
+- `Snapshot getSnapshot()`
+  Returns a snapshot of the current database state
+- `void withSnapshot(Consumer<Snapshot> snapshotFunction)`
+  Performs a given function with a snapshot of the current database state
+- `<ResultT> ResultT applySnapshot(Function<Snapshot, ResultT> snapshotFunction)`
+  Performs a given function with a snapshot of the current database state and
+  returns a result of its execution
+
+!!! note
+    Note that `withSnapshot` and `applySnapshot` methods destroy the snapshot
+    once the passed closure completes, compared to `getSnapshot`, which
+    disposes created snapshots only when TestKit is closed.
+    Therefore it is recommended to use first two methods if a large number
+    (e.g. more than a hundred) of snapshots needs to be created.
+
+### Time Oracle Testing
+
+The TestKit allows to use [Time Oracle][time-oracle] in integration tests if
+your service depends on it. To do that, TestKit should be created with
+[`TimeProvider`][testkit-time-provider].
+Its implementation [`FakeTimeProvider`][testkit-fake-time-provider] mocks the
+source of external data (current time) and therefore allows to manually
+manipulate time that is returned by time service. Note that it only supports
+time in UTC.
+
+??? note "Test example"
+    ```java
+    @Test
+    void timeOracleTest() {
+      ZonedDateTime initialTime = ZonedDateTime.now(ZoneOffset.UTC);
+      FakeTimeProvider timeProvider = FakeTimeProvider.create(initialTime);
+      try (TestKit testKit = TestKit.builder()
+          .withService(MyServiceModule.class)
+          .withTimeService(timeProvider)
+          .build()) {
+        // Create an empty block
+        testKit.createBlock();
+        // The time service have submitted its first transaction in `afterCommit`
+        // method, but it has not been executed yet
+        Optional<ZonedDateTime> consolidatedTime1 = getConsolidatedTime(testKit);
+        // No time available till the time service transaction is processed
+        assertThat(consolidatedTime1).isEmpty();
+
+        // Advance the time
```

Comment:
Probably "Advaces" if it is a verb. However, the meaning is vague.

Reply:
Is `Increase` better than `Advance`?

Reply:
> move forward
​[intransitive, transitive] (formal) to move forward to a later part of something; to move something forward to a later part 
> * Users advance through the program by answering a series of questions. advance something 
> * This button advances the hours and the red one advances the minutes in the display.

https://www.oxfordlearnersdictionaries.com/definition/english/advance_2

Not sure if it can be applied to what is, essentially, a clock.